src/cryptonote_protocol/cryptonote_protocol_handler.inl

```
@@ -902,6 +933,92 @@ namespace cryptonote
   }
   //------------------------------------------------------------------------------------------------------------------------
   template<class t_core>
+  int t_cryptonote_protocol_handler<t_core>::handle_notify_tx_pool_inv(int command, NOTIFY_TX_POOL_INV::request& arg, cryptonote_connection_context& context)
+  {
+    MLOG_P2P_MESSAGE("Received NOTIFY_TX_POOL_INV (" << arg.t.size() << " txes)");
+
+    // Create a list to hold transaction hashes missing in our local tx pool.
+    std::vector<crypto::hash> missing_tx_hashes;
+
+    // Iterate over each advertised transaction hash and check our pool and our requested tracker.
+    for (const auto &tx_hash : arg.t)
+    {
+      if (!context.can_process_additional_request())
```

P2P_MAX_IN_FLIGHT_REQUESTS=100 seems pretty low to just stop handling new tx hashes and silently ignore them. The daemon wouldn't be able to handle more than 100 new tx hashes it hasn't seen before per connection. I think instead it would make more sense to queue tx hashes for requesting later on... and the `check_tx_request_queue` could be altered to kick off requests that haven't been kicked off before (edit: in addition to the deadline/stale check)